kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationFilter.java

```diff
@@ -125,23 +131,31 @@ public CompletionStage<RequestFilterResult> onSaslAuthenticateRequest(short apiV
                         .setErrorMessage("Unexpected SASL request")
                         .setAuthBytes(request.authBytes());
```

Comment:
I know it's not your code, but this looks highly suspicious. It makes no sense to return the `authBytes` which the client sent in its request back to the client. I think we should be just leaving the `authBytes` unset in this case.

Reply:
Good spot.  I didn't notice that when I initially reviewed the submission.  From a quick look at https://datatracker.ietf.org/doc/html/rfc7628#page-9, it looks wrong.  Let separate this into a separate issue.